ec2/InstanceType.dhall

```diff
@@ -389,70 +410,268 @@ let alternatives =
                           )
                 , c5-4xlarge =
                     let cross-instances =
-                          [ InstanceType.m5-4xlarge, InstanceType.r5-4xlarge ]
+                          let lowClass =
+
+                              {-
+                                lowClass - all instance types that have lower ECU and lower network performance
+                                  -}
+                                [ InstanceType.m4-4xlarge ]
+
+                          in    [ InstanceType.m5-4xlarge
+                                , InstanceType.r5-4xlarge
+                                ]
+                              # ( if        _params.permitLowerClassCPU
+                                        &&  _params.permitWorseNetwork
+                                  then  lowClass
+                                  else  [] : List InstanceType
+                                )
 
                     let instancefamily =
-                          [ InstanceType.c5d-4xlarge, InstanceType.c5n-4xlarge ]
+                          let lowDiskBandwidth =
+
+                              {- lowDiskBandwidth - an instance type with low EBS disk bandwidth -}
+
+                                [ InstanceType.c5a-4xlarge ]
+
+                          in    [ InstanceType.c5d-4xlarge
+                                , InstanceType.c5n-4xlarge
+                                ]
+                              # ( if    _params.permitLowerEBSBandwidth
+                                  then  lowDiskBandwidth
+                                  else  [] : List InstanceType
+                                )
 
                     in    instancefamily
                         # ( if    _params.permitCrossInstanceFamily
-                            then    cross-instances
-                                  # ( if        _params.permitLowerClassCPU
-                                            &&  _params.permitWorseNetwork
-                                      then  [ InstanceType.m4-4xlarge ]
-                                      else  [] : List InstanceType
-                                    )
+                            then  cross-instances
+                            else  [] : List InstanceType
+                          )
+                , c5a-large =
+                    let cross-instances =
+                          let lowClass =
+
+                              {-
+                                lowClass - all instance types that have lower ECU and lower network performance
+                                  -}
+                                [ InstanceType.m4-large ]
+
+                          in    [ InstanceType.m5a-large
+                                , InstanceType.r5a-large
+                                ]
+                              # ( if        _params.permitLowerClassCPU
+                                        &&  _params.permitWorseNetwork
+                                  then  lowClass
```

Comment:
You need to have permitLowerEBSBandwidth here too because c5a.large has EBS bandwidth up to 3170 Mbps while m4.large has EBS bandwidth of 450 Mbps dedicated